phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -666,8 +681,8 @@ public void clearTableRegionCache(byte[] tableName) throws SQLException {
                 do {
                     HRegionLocation regionLocation = connection.getRegionLocation(
                             TableName.valueOf(tableName), currentKey, reload);
```

Comment:
@yanxinyi @dbwong are the locations in the regionLocations object always sorted ? If yes, then only it makes sense to compare if previous endKey is lesser than current endKey otherwise just comparing for equality will give us an answer if it is overlap.

Reply:
yes, this is the reason that we need to make sure the end key always increasing.